### PR TITLE
DOC: improved the docstring of pandas.Series.clip_lower

### PR DESCRIPTION
**Pull request by dcanones**
```
(pandas_dev) david@david-TM1604:~/repos/pandas/scripts$ python validate_docstrings.py pandas.Series.clip_lower

################################################################################
##################### Docstring (pandas.Series.clip_lower) #####################
################################################################################

Return copy of the input with values below given value(s) truncated.

It truncate values below a certain threshold. Threshold can be a single
value or an array, in the latter case it performs the truncation element-wise.

Parameters
----------
threshold : float or array_like
    Minimum value allowed. All values below threshold will be
    set to this value.
axis : int or string axis name, optional
    Align object with threshold along the given axis.
inplace : boolean, default False
    Whether to perform the operation in place on the data.

See Also
--------
clip : Return copy of the input with values below and over given thresholds values truncated.

Returns
-------
clipped : same type as input

Examples
--------
>>> s = pd.Series([3.816052, 7.969235, 6.903116, 2.532261, 5.015296])
>>> s
0    3.816052
1    7.969235
2    6.903116
3    2.532261
4    5.015296
dtype: float64

>>> s.clip_lower(4)
0    4.000000
1    7.969235
2    6.903116
3    4.000000
4    5.015296
dtype: float64

>>> t = [4,8,7,2,5]
>>> t
[4, 8, 7, 2, 5]

>>> s.clip_lower(t)
0    4.000000
1    8.000000
2    7.000000
3    2.532261
4    5.015296
dtype: float64

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.clip_lower" correct. :)
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5661,22 +5661,58 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
         ----------
         threshold : float or array_like
+            Minimum value allowed. All values below threshold will be
+            set to this value.
         axis : int or string axis name, optional
             Align object with threshold along the given axis.
         inplace : boolean, default False
-            Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+            Whether to perform the operation in place on the data.
```
Comment (datapythonista) [inline pandas/core/generic.py]: Please, keep the versionadde

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5661,22 +5661,58 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
+        clip : Return copy of input with values below and above thresholds truncated.
 
         Returns
         -------
         clipped : same type as input
+
+        Examples
+        --------
+        >>> s = pd.Series([3.816052, 7.969235, 6.903116, 2.532261, 5.015296])
+        >>> s
```
Comment (datapythonista) [inline pandas/core/generic.py]: I think it'll be much easier for people to understand the example with integers (e.g. `[1, 2, 3, 4]`). Not sure if I'm missing some reason for using this numbers with so many decimals.

**Review by datapythonista — COMMENTED** [review]
Looks great, couple of comments

**Review by jreback — APPROVED** [review]

**Comment by dcanones** [comment]
Why is Travis and Appveyor failing?

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5661,22 +5661,59 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
             Align object with threshold along the given axis.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
 
         See Also
         --------
-        clip
+        clip : Return copy of input with values below and above thresholds truncated.
```
Comment (datapythonista) [inline pandas/core/generic.py]: don't we have a clip_upper too, that it's worth listing here? Also, I didn't check, but I understand that the clip listed here is `pandas.clip`, and this is a method of `Series` and `DataFrame`. Is there a `Series.clip` too? We may want to list it here as well then.

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5661,22 +5661,59 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
         ----------
         threshold : float or array_like
+            Minimum value allowed. All values below threshold will be
+            set to this value.
         axis : int or string axis name, optional
             Align object with threshold along the given axis.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
```
Comment (datapythonista) [inline pandas/core/generic.py]: The dot should go in the previous line. If the validation script complaints about it, ignore the message.

File: pandas/core/generic.py [hunk 5]
```diff
@@ -5661,22 +5661,59 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
+
+        >>> s.clip_lower(4)
+        0    5
+        1    6
+        2    7
+        3    8
+        4    9
+        dtype: int64
+
+        >>> t = [4,8,7,2,5]
```
Comment (datapythonista) [inline pandas/core/generic.py]: Missing spaces for PEP-8

Also, what does `t` stands for? can you use a more descriptive name please?

File: pandas/core/generic.py [hunk 6]
```diff
@@ -5661,22 +5661,59 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
+        >>> t
+        [4, 8, 7, 2, 5]
```
Comment (datapythonista) [inline pandas/core/generic.py]: Personally I don't think this adds value, I'd take it out and make the dosctring a bit shorter.

File: pandas/core/generic.py [hunk 7]
```diff
@@ -5661,22 +5661,59 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
+        4    9
+        dtype: int64
+
+        >>> s.clip_lower(4)
+        0    5
+        1    6
+        2    7
+        3    8
+        4    9
+        dtype: int64
```
Comment (datapythonista) [inline pandas/core/generic.py]: In this example, `clip_lower` returns the input as it is. May be a more meaningful example?

File: pandas/core/generic.py [hunk 8]
```diff
@@ -5661,22 +5661,59 @@ def clip_lower(self, threshold, axis=None, inplace=False):
...
+
+        >>> t = [4,8,7,2,5]
+        >>> t
+        [4, 8, 7, 2, 5]
+
+        >>> s.clip_lower(t)
+        0    5
+        1    8
```
Comment (datapythonista) [inline pandas/core/generic.py]: May be a sentence before this explaining what this example shows would be useful.

**Review by datapythonista — COMMENTED** [review]
Nice PR, added some comments.

**Comment by datapythonista** [comment]
Don't worry about Travis and the CI, some times they fail because of something merged before your PR.

**Comment by dcanones** [comment]
Reviewer requests added

File: pandas/core/generic.py (hunk 3 above)
Comment (dcanones) [reply pandas/core/generic.py]: Yes, there is Series.clip and DataFrame.clip and Series.clip methods. I listed both in the last commit

**Comment by datapythonista** [comment]
lgtm

**Review by jreback — APPROVED** [review]

**Review by TomAugspurger — COMMENTED** [review]
Added examples for DataFrame.

Can someone lmk if my explanation of `theshold` and `axis` makes sense?

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks @dcanones

**Pull request merged**